docker/codex.Dockerfile

```diff
@@ -1,14 +1,18 @@
-FROM nimlang/nim:1.6.10-alpine AS builder
+FROM ubuntu:lunar-20230415 AS builder
+RUN apt-get update && apt-get install -y git cmake curl make bash lcov build-essential nim
+RUN echo 'export NIMBLE_DIR="${HOME}/.nimble"' >> "${HOME}/.bash_env"
+RUN echo 'export PATH="${NIMBLE_DIR}/bin:${PATH}"' >> "${HOME}/.bash_env"
+
 WORKDIR /src
-RUN apk update && apk add git cmake curl make git bash linux-headers
 COPY . .
 RUN make clean
-RUN make update
-RUN make NIM_PARAMS="-d:disableMarchNative"
+RUN make -j4 update
```

Comment:
we want this to be `-j` to be parametrizable, probably thru env variables.

Reply:
Default value is 4 and we can adjust it using `build-arg`
```bash
docker build \
  --progress plain \
  --build-arg MAKE_PARALLEL=8 \
  --file docker/codex.Dockerfile .
```

```
#12 [builder 8/9] RUN make -j 8 update
#12 0.355 Creating nimbus-build-system.paths
```